app/models/cas_application_fields.rb

```diff
@@ -17,4 +21,26 @@ def declaration_choices
       end
     end.compact.to_h
   end
+
+  def reason_justification_must_be_populated_if_level_1_selected
+    return if reason_justification.present?
+    return unless project.project_datasets.any? do |pd|
```

Comment:
I think you've already got a direct association on project i.e 
`project.project_dataset_levels.where(access_level_id: Lookups::AccessLevel.find_by(value: 1)).count.positive?`

Reply:
project_dataset_levels association doesn’t work with the project helper that we are using to build all possible levels in memory in the form. Therefore, the levels and associated project_datasets that we want to evaluate here aren't persisted at the point of validation.